stdlib/public/core/StringBridge.swift

```diff
@@ -234,35 +243,143 @@ internal func _bridgeCocoaString(_ cocoaString: _CocoaString) -> _StringGuts {
+  if cls == __StringStorage.self {
+    return String(_unsafeUncheckedDowncast(
+      cocoaString, to: __StringStorage.self).asString._guts)
+  }
+  return String(_unsafeUncheckedDowncast(
+    cocoaString, to: __SharedStringStorage.self).asString._guts)
+}
+
+@_effects(releasenone)
+public //SPI(Foundation)
+func _bridgeTaggedCocoaString(_ cocoaString: AnyObject) -> String? {
+  #if !(arch(i386) || arch(arm))
+  if _isObjCTaggedPointer(cocoaString) {
+    return String(_StringGuts(_SmallString(taggedCocoa: cocoaString)))
+  }
+  #endif
+  return nil
+}
+
+@_effects(releasenone)
+public //SPI(Foundation)
+func _bridgeShortCFString(
+  _ cocoaString: AnyObject, _ length: Int) -> String? {
+  if length <= _SmallString.capacity {
+    if let smol = _SmallString(nonTaggedCocoa: cocoaString, length: length) {
+      return String(_StringGuts(smol))
+    }
+  }
+  return nil
+}
+
+@_effects(releasenone) // @opaque
+internal func _bridgeOwnedCocoaString(_ cocoaString: __owned _CocoaString) ->
+  _StringGuts {
+    /*switch CocoaStringKind(cocoaString) {
+    case .storage:
+      return _unsafeUncheckedDowncast(
+        cocoaString, to: __StringStorage.self).asString._guts
+    case .shared:
+      return _unsafeUncheckedDowncast(
+        cocoaString, to: __SharedStringStorage.self).asString._guts
+    case .cf:
+      let length = _stdlib_binary_CFStringGetLength(cocoaString)
+      
+      //The address of an NSString constant can be semantically meaningful
+      //e.g. NSRunLoop modes, so we can't be clever like trying a SmallString
+      //Similarly, NSString subclasses may do odd things, so we should only try
+      //to eager-bridge regular CF strings.
+      //TODO: access constant NSString contents directly, since they're ABI
+      //TODO: if https://bugs.swift.org/browse/SR-9988 happens, tell swiftc
+      //that this object is immortal if constant
+      if length <= _SmallString.capacity {
+        //We *may* not be able to form a SmallString from this, but above 15 we
+        //definitely can't. It's worth trying because eager bridging is almost
+        //always good if we don't have to allocate for it.
+        if let smol = _SmallString(nonTaggedCocoa: cocoaString, length: length) {
+          return _StringGuts(smol)
+        }
+      }
+      return _bridgeNonSwiftCocoaString(cocoaString, length: length, nonCF: false)
+    default:
+      let length = _stdlib_binary_CFStringGetLength(cocoaString)
+      return _bridgeNonSwiftCocoaString(cocoaString, length: length, nonCF: true)
+    }*/
+fatalError("should not be called")
+}
+
+@usableFromInline
+@_effects(releasenone) // @opaque
+internal func _bridgeCocoaString(_ cocoaString: _CocoaString)
+  -> _StringGuts {
+    return _bridgeOwnedCocoaString(cocoaString)
 }
 
 extension String {
   public // SPI(Foundation)
   init(_cocoaString: AnyObject) {
-    self._guts = _bridgeCocoaString(_cocoaString)
+    self._guts = _bridgeOwnedCocoaString(_cocoaString)
```

Comment:
Try Jordan's hack to generate a call to the new codepath: self._guts = (_cocoaString as Any as! String)._guts